### DISCUSSION TITLE
search functionality with ontology

### DISCUSSION THREAD

**Discussion by BerkvensNick** [soilwise-he/harvesters#86] [Announcements]
@pvgenuchten this was a remark that came up from @KathiSchleidt  in a WP5 meeting:

What's the planned search approach? Utilizing an underlying Ontology that could sort the synonyms via sameAs, or something more towards Lucene?

not sure if the issue is relevant to the harvester, if not, please move to the relevant repo

**Comment by pvgenuchten** [comment]
We will for sure provide a vocabulary in the backend which facilitates reasoning on language labels, synonyms, sameas, maybe even narrower/broader.

The indexing mechanism could match relevant keywords based on that from the above graph

At search time the options are limited for keyword matching, using current technology, elastic/solr may be better here. for the generic search (not keyword filter) parsing the querystrung using nlp/llm would provide additional value, eg: “soil data of Flanders by ilvo” -> theme:soil type:dataset location:Flanders organization:ilvo
